src/operators/join/join.cc

```diff
@@ -253,17 +262,29 @@ void Join::ProbeHashTableBlock(
         (uint32_t *)malloc(sizeof(uint32_t) * chunk_length);
     auto *joined_right_indices =
         (uint32_t *)malloc(sizeof(uint32_t) * chunk_length);
-
+    auto indices_length = chunk_length;
+    // TODO (suryadev) : As of now, we have two separate cases with
+    //  nearly same code with conditionals difference for faster loop execution
+    // as this forms one of the core code block for most of the queries with
+    // joins.
     if (filter_data != nullptr) {
       for (std::size_t row = 0; row < chunk_length; row++) {
         if (arrow::BitUtil::GetBit(filter_data, row)) {
           auto key_value_pair =
               hash_tables_[join_id]->find(left_join_chunk_data[row]);
           if (key_value_pair != hash_table_end) {
-            joined_left_indices[num_joined_indices] = row + offset;
-            joined_right_indices[num_joined_indices] =
-                key_value_pair->second.index;
-            ++num_joined_indices;
+            for (auto record_id : key_value_pair->second) {
+              if (indices_length <= num_joined_indices) {
+                indices_length << 1;
```

Comment:
`indices_length <<= 1;` ? Do you mean to shift the `indicies_length`?

Reply:
Oop! Thank you!